### PR TITLE
SERVICES-2122: duplicated transfer value only logs

### PR DESCRIPTION
**Pull request by bogdan-rosianu**
## Reasoning
- with the latest protocol release, misleading duplicated events can be returned in some edge cases
  
## Proposed Changes
- until a new protocol release, alter the logs to avoid misleading information

## How to test
- `http://localhost:3001/transactions/9577b1e88608495a5dc259dc2e313730f44871b859f575bc6e14f380df04ac15` when running on devnet should have the first topic of the event "transferValueOnly" with data field "QmFja1RyYW5zZmVy" set to "0"
- other transactions logs / events should not be affected

### CODE REVIEW

File: src/endpoints/transactions/transaction.get.service.ts [hunk 1]
```diff
@@ -126,6 +129,63 @@ export class TransactionGetService {
...
+    }
+
+    if (this.hasDuplicateDataWithMatchingTopics(transactionDetailed.logs.events)) {
+      this.setTopicAtIndexInEventWhereIdentifier("0", 0, transactionDetailed.logs.events, "transferValueOnly");
+    }
+  }
+
+  private hasDuplicateDataWithMatchingTopics(events: TransactionLogEvent[]): boolean {
+    const identifier = "transferValueOnly";
+    const dataValue1 = BinaryUtils.base64Encode("BackTransfer");
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply src/endpoints/transactions/transaction.get.service.ts]: done

**Review by cfaur09 — DISMISSED** [review]

**Review by tanghel — APPROVED** [review]
